chain/chunks/src/chunk_cache.rs

```diff
@@ -143,39 +141,41 @@ impl EncodedChunksCache {
     }
 
     pub fn remove(&mut self, chunk_hash: &ChunkHash) -> Option<EncodedChunksCacheEntry> {
-        match self.encoded_chunks.entry(chunk_hash.clone()) {
-            Occupied(entry) => {
-                let entry = entry.remove_entry().1;
-                self.remove_incomplete_chunk(entry.header.prev_block_hash(), chunk_hash);
-                Some(entry)
-            }
-            Vacant(_) => None,
+        if let Some(entry) = self.encoded_chunks.remove(chunk_hash) {
+            self.remove_chunk_from_incomplete_chunks(entry.header.prev_block_hash(), chunk_hash);
+            Some(entry)
+        } else {
+            None
         }
     }
 
     // Remove the chunk from the `incomplete_chunks` map. This is an internal function.
     // Use `mark_entry_complete` instead for outside calls
-    fn remove_incomplete_chunk(&mut self, prev_block_hash: CryptoHash, chunk_hash: &ChunkHash) {
-        if let Some(mut chunks) = self.incomplete_chunks.remove(&prev_block_hash) {
-            chunks.remove(chunk_hash);
-            if !chunks.is_empty() {
-                self.incomplete_chunks.insert(prev_block_hash, chunks);
+    fn remove_chunk_from_incomplete_chunks(
+        &mut self,
+        prev_block_hash: CryptoHash,
+        chunk_hash: &ChunkHash,
+    ) {
+        if let Occupied(mut entry) = self.incomplete_chunks.entry(prev_block_hash) {
+            entry.get_mut().remove(chunk_hash);
+            if entry.get().is_empty() {
+                entry.remove();
             }
         }
     }
 
     /// Insert if entry does not exist already
-    pub fn try_insert(&mut self, chunk_hash: ChunkHash, header: &ShardChunkHeader) {
-        self.get_or_insert_from_header(chunk_hash, header);
+    pub fn try_insert(&mut self, header: &ShardChunkHeader) {
+        self.get_or_insert_from_header(header);
     }
 
     // Create an empty entry from the header and insert it if there is no entry for the chunk already
     // Return a mutable reference to the entry
-    pub fn get_or_insert_from_header(
+    fn get_or_insert_from_header(
         &mut self,
-        chunk_hash: ChunkHash,
         chunk_header: &ShardChunkHeader,
     ) -> &mut EncodedChunksCacheEntry {
+        let chunk_hash = chunk_header.chunk_hash();
         if !self.encoded_chunks.contains_key(&chunk_hash) {
```

Comment:
This should be inside the if.
```rust
        self.encoded_chunks
            .entry(chunk_hash)
            .or_insert_with(|| EncodedChunksCacheEntry::from_chunk_header(chunk_header.clone()))
```

Then it can be written more simply as.
```rust
        self.encoded_chunks
            .insert(chunk_hash, EncodedChunksCacheEntry::from_chunk_header(chunk_header.clone()))
```

Reply:
chunk_hash is used in line 179

`.entry.or_insert_with` saves a call to construct the EncodedChunksCacheEntry object if the entry already exists.